src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.CompilationAnalyzer.cs

```diff
@@ -41,9 +41,9 @@ public void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context)
             public static void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
             {
                 var declDiagnostics = context.SemanticModel.GetDeclarationDiagnostics(cancellationToken: context.CancellationToken);
-                ReportDiagnostics(declDiagnostics, context.ReportDiagnostic, IsSourceLocation, s_declaration);
-
                 var bodyDiagnostics = context.SemanticModel.GetMethodBodyDiagnostics(cancellationToken: context.CancellationToken);
+
+                ReportDiagnostics(declDiagnostics, context.ReportDiagnostic, IsSourceLocation, s_declaration);
```

Comment:
This change might address duplicate diagnostics in presence of cancellation, how does it handle missing diagnostics, which seem to be the reason for all failures?

Reply:
This change seems desirable regardless of whether it addresses a test failure or not. We should not report half of diagnostics when cancelled. All or nothing.

---
In reply to: [473253945](https://github.com/dotnet/roslyn/pull/46974#discussion_r473253945) [](ancestors = 473253945)

Reply:
Yep, no argument about this being a desirable change.